packages/key-readmodel-writer/src/keyConsumerServiceV1.ts

```diff
@@ -1,15 +1,47 @@
 import { KeyCollection } from "pagopa-interop-commons";
-import { AuthorizationEventEnvelopeV1 } from "pagopa-interop-models";
+import {
+  AuthorizationEventEnvelopeV1,
+  fromKeyV1,
+  toReadModelKey,
+} from "pagopa-interop-models";
 import { match } from "ts-pattern";
 
 export async function handleMessageV1(
   message: AuthorizationEventEnvelopeV1,
-  _keys: KeyCollection
+  keys: KeyCollection
 ): Promise<void> {
-  match(message)
+  await match(message)
+    .with({ type: "KeysAdded" }, async (message) => {
+      const keysToAdd = message.data.keys.map((keyEntry) => keyEntry.value);
+
+      for (const key of keysToAdd) {
+        if (key) {
+          await keys.updateOne(
+            {
+              "data.kid": key.kid,
+              "metadata.version": { $lt: message.version },
+            },
+            {
+              $set: {
+                data: toReadModelKey(fromKeyV1(key)),
+                metadata: {
+                  version: message.version,
+                },
+              },
+            },
+            { upsert: true }
+          );
+        }
+        await Promise.resolve();
+      }
+    })
+    .with({ type: "KeyDeleted" }, async (message) => {
+      await keys.deleteOne({
+        "data.kid": message.data.keyId,
+        "metadata.version": { $lt: message.version },
+      });
+    })
     .with(
-      { type: "KeysAdded" },
-      { type: "KeyDeleted" },
       { type: "KeyRelationshipToUserMigrated" },
       { type: "ClientAdded" },
       { type: "ClientDeleted" },
```

Comment:
@galales should we handle `ClientDeleted` here as well? If we want to delete all its keys, how do we retrieve them? The event only holds `clientId`, but if we query the `clients` collection it might have already been deleted (by the `client-readmodel-writer`).

Reply:
Also, are there any other events that need to be processed?

Reply:
yes, `ClientDeleted` should be handled to delete all its keys 👍 
We can directly delete documents in `keys` collection by `clientId` field

Reply:
Actually, there is no `clientId` inside a `key`.

Reply:
Updated (in a previous PR, `clientId` has been added as a property of `key`)